codec/src/main/java/io/netty/handler/codec/DefaultHeaders.java

```diff
@@ -79,19 +119,47 @@ public T get(T name, T defaultValue) {
     }
 
     @Override
-    @SuppressWarnings("unchecked")
     public T getAndRemove(T name) {
```

Comment:
The code was copied from `remove0()`, originally to test with `remove0()` not having to incur the overhead of tracking the value. I will consolidate the two implementations and run a perf test.

As I understand it, with the first loop the `entries` array itself may be updated, but the second loop starts when it is not necessary to update the `entries` array.